### PR TITLE
[LoongArch] Add relaxDwarfLineAddr and relaxDwarfCFA to handle the mutable label diff in dwarfinfo

### PR DESCRIPTION
**Pull request by MQ-mengqing**
When linker-relaxation is enabled, part of the label diff in dwarfinfo cannot be computed before static link. Refer to RISCV, we add the relaxDwarfLineAddr and relaxDwarfCFA to add relocations for these label diffs. Calculate whether the label diff is mutable. For immutable label diff, return false and do the other works by its parent function.

### CODE REVIEW

**Comment by MQ-mengqing** [comment]
Add: @SixWeining @MaskRay @wangleiat @xen0n

File: llvm/lib/Object/RelocationResolver.cpp [hunk 1]
```diff
@@ -879,9 +885,11 @@ uint64_t resolveRelocation(RelocationResolver Resolver, const RelocationRef &R,
 
       if (GetRelSectionType() == ELF::SHT_RELA) {
         Addend = getELFAddend(R);
-        // RISCV relocations use both LocData and Addend.
+        // RISCV, LoongArch relocations use both LocData and Addend.
```
Comment (SixWeining) [inline llvm/lib/Object/RelocationResolver.cpp:888]: How about "LoongArch and RISCV ..." ?

And sort alphabetically:
```
        if (Obj->getArch() != Triple::loongarch32 &&
            Obj->getArch() != Triple::loongarch6464 &&
            Obj->getArch() != Triple::riscv32 &&
            Obj->getArch() != Triple::riscv64)
```

File: llvm/test/DebugInfo/LoongArch/dwarf-loongarch-relocs.ll [hunk 2]
```diff
@@ -0,0 +1,128 @@
+; RUN: llc --filetype=obj --mtriple=loongarch64 --mattr=-relax %s -o %t.o
+; RUN: llvm-readobj -r %t.o | FileCheck -check-prefixes=RELOCS-BOTH,RELOCS-NORL %s
```
Comment (SixWeining) [inline llvm/test/DebugInfo/LoongArch/dwarf-loongarch-relocs.ll:2]: ``--check-prefixes``

File: llvm/test/DebugInfo/LoongArch/dwarf-loongarch-relocs.ll [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+; RUN: llvm-objdump --source %t.o | FileCheck --check-prefix=SOURCE %s
+; RUN: llvm-dwarfdump --debug-info --debug-line %t.o | FileCheck -check-prefix=DWARF %s
```
Comment (SixWeining) [inline llvm/test/DebugInfo/LoongArch/dwarf-loongarch-relocs.ll:4]: `--check-prefix`

**Review by SixWeining — APPROVED** [review]
LGTM. Thanks.

File: llvm/test/DebugInfo/LoongArch/lit.local.cfg [hunk 4]
```diff
@@ -0,0 +1,2 @@
+if not "LoongArch" in config.root.targets:
```
Comment (MaskRay) [inline llvm/test/DebugInfo/LoongArch/lit.local.cfg:1]: `not in`

**Review by MaskRay — APPROVED** [review]

**Pull request merged**